Add Grid.prototype.clearAllFilters() method.

amyboyd opened the pull request:
I had a use-case where I needed a button that clears all filters applied to all columns. This pull request adds a method to the Grid class, `clearAllFilters()`, which does that.

PaulL1 commented:
This looks good.  A few comments:
1.  The test has a header of sorting - copy paste?
2.  You're clearing the condition and the flags as well.  The filter should become inactive when you remove the term, perhaps people would rather retain the condition?  Should this behaviour maybe be the default or at least switchable? 
3.  It seems you'd explicitly intend to have people use this on the api, perhaps it would make sense to add:
  `    self.api.registerMethod( 'core', 'clearAllFilters', this.clearAllFilters );`
to the top of the grid.js?

amyboyd commented:
Thanks for the feedback @PaulL1. I've changed the method signature to `Grid.prototype.clearAllFilters = function(refreshRows = true, clearConditions = false, clearFlags = false)`, added the `registerMethod` call, and fixed the copy-paste mistake. The method has 100% unit test coverage too.

PaulL1 commented:
Very tidy.  Thanks for the contribution.
